apollo-router/src/plugins/headers.rs

```diff
@@ -37,6 +37,15 @@ use crate::SubgraphRequest;
 
 register_plugin!("apollo", "headers", Headers);
 
+#[derive(Clone, JsonSchema, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+struct HeadersLocation {
```

Comment:
I think we need to mark this as being non-exhaustive:

```
#[non_exhaustive]
```

Reply:
Is it important to mark it as `non_exhaustive` even if the struct is private ?

Reply:
No it isn't. Ignore me.